### DISCUSSION TITLE
Increase in zsh startup time when loading zi pack for pyenv

### DISCUSSION THREAD

**Discussion by meinzer1899** [z-shell/zi#298] [Q&A] [answered]
When I add
`zi pack for pyenv` from [z-shell/pyenv](https://github.com/z-shell/pyenv), zsh startup time increases significantly:

![image](https://github.com/z-shell/zi/assets/44314956/3dc47d6d-2f33-4736-a685-3d3a6a53df68)

Do you have an idea?

**Comment by ss-o** [comment]
I will do triage on this one :+1: Are these startup times zsh with **no pyenv** compared to zsh **with pyenv**? :thinking:

**Reply by meinzer1899** [reply]
Thanks. Yes exactly. Increase of about 100 ms when using pyenv (~120 ms w/o pyenv, ~220 ms with).

**Comment by ss-o** [comment, marked as answer]
> Thanks. Yes exactly. Increase of about 100 ms when using pyenv (~120 ms w/o pyenv, ~220 ms with).

The [z-shell/pyenv](https://github.com/z-shell/pyenv) was created to speed up pyenv initialization, that being said to verify it, the **Zsh + regular pyenv** and **Zsh + [z-shell/pyenv](https://github.com/z-shell/pyenv)** should be compared. 

Could you confirm, that you compared regular installation of pyenv versus z-shell/pyenv?

**Reply by meinzer1899** [reply]
I did not; I compared things that weren't related. Your statement makes sense. Thanks @ss-o

**Reply by ss-o** [reply]
I still do triage for possible improvements :dart: 

If you need additional information on any topic/repository or just have questions feel free to get in touch :handshake:
